include/rmm/mr/device/callback_memory_resource.hpp

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright (c) 2020-2021, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <rmm/mr/device/device_memory_resource.hpp>
+
+#include <cstddef>
+#include <functional>
+#include <utility>
+
+namespace rmm::mr {
+
+/**
+ * @brief Callback function type used by callback memory resource for allocation.
+ *
+ * The signature of the callback function is:
+ *   `void* allocate_callback_t(std::size_t bytes, cuda_stream_view stream, void* arg);
+ *
+ * `bytes` specifies the number of bytes to allocate. `stream` is the CUDA stream
+ * on which the memory is used. Additional information to the callback may be passed
+ * via `arg`. The function returns the pointer to the allocated memory.
```

Comment:
Update the `deallocate_callback_t` similarly. 

```suggestion
 * Returns a pointer to an allocation of least `bytes` usable immediately on `stream`. The stream-ordered 
 * behavior requirements are identical to `device_memory_resource::allocate`. 
 * 
 * Additional context to the callback may be passed via `arg` that is provided at construction of the `callback_memory_resource`.  
```

Reply:
Done